### ISSUE TITLE
Separate image server

### ISSUE THREAD

**Issue by jechols** [open-oni/open-oni#16]
- Allow user to specify image server URL
- Centralize all image URL generation
- Use IIIF-compatible paths

**Comment by karindalziel** [comment]
Settings to configure URL - localhost vs server.

**Comment by karindalziel** [comment]
Leaving this out of the January code sprint unless someone else can pick it up.

**Comment by edsu** [comment]
So is the plan for an open-oni installation to require an IIIF compliant image server, and for the current image tiling code in the existing chronam to be removed? Or has the discussion not proceeded that far yet?

**Comment by karindalziel** [comment]
@edsu I think the idea was to make it so you could use either the built in code OR an image server.

**Comment by edsu** [comment]
Ok, that makes sense. I think that might something I can help out with. @kestlund let me know that a new sprint was starting next week. Are you all doing a phone call or anything to kick things off, or has that already happened?

**Comment by karindalziel** [comment]
@edsu Yes! There's a call this Friday. See messages on chronam Dev list:

http://listserv.loc.gov/cgi-bin/wa?A2=ind1601&L=chronam-users&T=0&P=556

http://listserv.loc.gov/cgi-bin/wa?A2=ind1601&L=chronam-users&T=0&P=661

In case those don't work, 

Open-ONI sprint planning:
Friday January 15 @ 11am Pacific (12pm Mountain, 1pm Central, 2pm Eastern)
Link to Google Hangout: https://hangouts.google.com/call/ig66jkyvx235jidllryvadvny4a

And this planning link on the wiki should have the rest (eventually): 

https://github.com/open-oni/open-oni/wiki/January-2016-Code-Sprint

**Comment by jechols** [comment]
Actually, we (UO) were hoping to remove all image-related code in order to reduce system requirements, remove some mental overhead, and reduce long-term maintenance issues.  Ideally, we'd be able to drop everything related to image serving, and just make sure our docker strategy gives people an easy way to stand up the IIIF server.  RAIS is already very docker-friendly, but of course I'm pretty biased on that one - but I've also heard Loris is easy to stand up with docker, too.

As for the sprint, I figured I'd tackle this one since I already did the work for the UO instance of chronam.  This is the area I've spent the most time in chronam, so I could certainly look at other pieces of the codebase, but I'm not sure my time will be as productive anywhere else.

**Comment by kestlund** [comment]
We had two primary goals with this issue:
1) To provide alternative path to using built in image server
2) Provide a path to IIIF
3) To retain ease for institutions to install (this is where we did not finalize the OR vs. the complete separation of all image-related code) and I think has led to the confusion here

Are there are IIIF issues that we should add to make this content more compliant?

As for the issue tasks, the removal from the sprint milestone created a bit of confusion. Perhaps good to discuss strategies during sprint planning.

@jechols @edsu @karindalziel

**Comment by jechols** [comment]
Heya, @kestlund!  I suspect you're correct about the confusion - I may have forgotten about not finalizing a plan here.

For UO, we are glad to drop the internal code, but I can understand the concerns related to ease of install.  From my perspective, the app is already a bit of a monster to install, and adding a simple server like RAIS (shameless plug :P) shouldn't make it a whole lot worse.  Additionally, any code we keep becomes extra work for devs in the long run.  In a perfect world, we would ditch everything not absolutely necessary, just to make it easier for people to jump in and code things up.  Resources being what they are, the more we can do to reduce ramp-up time each time we do work, the better we'll be long-term.

That said, we could certainly do this in stages, leaving the existing code in until I can convince everybody how brilliant my plans are :)

- Refactor all image URL generation to be IIIF-compliant
- Add an item to settings.py to allow for overriding the base IIIF path for tiles and static thumbnails (I want these separate because it's a huge win to cache thumbnails vs. the dynamic tiles)
- Add stuff to the example apache config to demonstrate use of a proxied IIIF server
- Rip out all the image-related code if everybody decides to go that route

The first three steps could be done in the sprint, I'm sure, as we've already done the majority of that work here at UO.

**Comment by edsu** [comment]
Just a quick update that I'm working on a branch over [here](https://github.com/edsu/open-oni/tree/iiif) to get all the images coming out of IIIF (using rais-image-server) that the open-oni deployment now depends on.

I agree with @jechols's assessment that the installation isn't exactly easy to begin with; and adding an iiif image server into the mix doesn't change things substantially. Also, the app performs *much* better using rais-image-server, and I imagine it will with iip server too.

For the record I stayed neutral about this decision when it came up at the end of the sprint. I figured it wasn't for me to decide since UMD has no plans to use open-oni at present. Stanford is funding UMD to work on making open-oni IIIF compatible, and it seemed like @jechols plan was a viable path to doing that.
